docs_src/security/Ch-Docker-Swarm-SecuringDeviceServices.md

```diff
@@ -88,133 +120,107 @@ Here is the inverse of the previous yml block. This gets added to the device ser
 
 These work because we set the 'hostname = worker-node' in the Vagrantfile.
 
-### Adding Host Mounted Volumes
+### Volumes
 
-In docker swarm bind mount volumes need to be explicitly defined in the stack yml file.  Below are the first three bind mount volume definitions these directories must be created on the host before the stack file can be ran.
+In secure mode, a shared volume, such as might be provided by NFS or GlusterFS,
+is needed to distribute the secret store token for the device service.
+Volumes defined using the default Docker Swarm driver are local to the node,
+and cannot be remotely accessed.
 
-Note that this is only an example.  In a production deployment you would want to use a network filesystem or create the shared volumes between containers.
+For simplicity, we will just make a secrets volume that holds all of
+the microservice secrets for the primary node.
+A production deployment should have a secrets volume per-microservice.
 
-```yaml
-  secrets-volume:
-    driver: local
-    driver_opts:
-      o: bind
-      type: none
-      device: /tmp/edgex/secrets/
-  secrets-ca-volume:
-    driver: local
-    driver_opts:
-      o: bind
-      type: none
-      device: /tmp/edgex/secrets/ca/
-  edgex-consul:
-    driver: local
-    driver_opts:
-      o: bind
-      type: none
-      device: /tmp/edgex/secrets/edgex-consul/
-...
+Here is an example substitution:
+
+```text
+<     - /tmp/edgex/secrets/core-command:/tmp/edgex/secrets/<service-name>:ro,z
```

Comment:
The original docker-compose maps the specific service's folder, for example, /tmp/edgex/secrets/core-command in the same place in both the host and the docker container.  Swarm, for obvious reasons, doesn't like host path mounts at all.  For simplicity, I've just created a docker volume edgex-secrets and mount the root of it at /tmp/edgex/secrets.

The intent is to show you can use a network overlay to distribute device services so I've simplified to avoid pulling my remaining hair out.